msm/sde/sde_hw_ctl.c

```diff
@@ -648,10 +648,17 @@ static inline u32 sde_hw_ctl_get_intf(struct sde_hw_ctl *ctx)
 
 	c = &ctx->hw;
 	ctl_top = SDE_REG_READ(c, CTL_TOP);
+	if (!ctl_top)
+		goto end;
 
-	intf_active = (ctl_top > 0) ?
-		BIT(ctl_top - 1) : 0;
+	if (IS_SDM630_TARGET(hwversion) ||
+	    IS_SDM660_TARGET(hwversion) ||
+	    IS_SDM845_TARGET(hwversion))
+		intf_active = (ctl_top >> 4) & 0xf;
+	else
+		intf_active = BIT(ctl_top - 1);
```

Comment:
This original read seems wrong anyway, Look at the setters: `sde_hw_ctl_intf_cfg` writes it left-shifted by 4 regardless of the platform. Are you sure techpack isn't just bugging out here?

Regardless, this 4.19 techpack _only_ uses it in `sde_rm` to check if it's non-zero, and it seems to rely on `CTL_0` starting at 1 (so intf `0` doesn't exist).